src/gt4py/frontend/gtscript_frontend.py

```diff
@@ -103,6 +105,101 @@ def __init__(self, name, data_type, message=None, *, loc=None):
+            partial_sizes = [0] + list(itertools.accumulate(sizes))
+            offset = partial_sizes[line - 1] + col - 1
+            index = source[offset:].find("(")
+            assert index > 0
+            offset += index + 1
+            start, scope = offset, 1
+            while scope > 0:
+                if source[offset] in ("(", "["):
+                    scope += 1
+                elif source[offset] in (")", "]"):
+                    scope -= 1
+                offset += 1
+
+            return source[start : offset - 1]
+
+        def is_valid_specifier(spec):
+            # None regions were already removed
+            if not isinstance(spec, Iterable):
+                return False
+            for axis in spec:
+                if axis is None:
+                    continue
+                try:
+                    for endpt in ("start", "stop"):
+                        if not isinstance(axis[endpt][0], gt_definitions.Interval):
+                            return False
+                        elif not isinstance(axis[endpt][1], int):
+                            return False
+                except:
+                    return False
+            return True
+
+        item_contexts = [x.context_expr for x in node.items]
+        assert all(isinstance(x, ast.Call) for x in item_contexts)
+
+        func_ids = [x.func.id for x in item_contexts]
+
+        if "interval" in func_ids:
+            # Store in the interval node
+            index = func_ids.index("interval")
+        elif "computation" in func_ids:
+            # Store in the computation node and apply to all intervals inside that
+            index = func_ids.index("computation")
+        else:
+            raise GTScriptSymbolError("With node must have either interval or computation")
+
+        if "region" in func_ids:
+            arg_index = func_ids.index("region")
+            region_node = item_contexts[arg_index]
+            code = (
+                find_region_source(self.source, region_node.lineno, region_node.col_offset).rstrip(
+                    " \n,"
+                )
+                + ","
+            )
+            specifiers = (tuple(item) for item in eval(f"({code})", self.context))
+
+            # Remove None objects (these are skipped regions)
+            specifiers = tuple(filter(lambda x: x is not None, specifiers))
+
+            for spec in specifiers:
+                if not is_valid_specifier(spec):
+                    raise GTScriptSymbolError(f"Not a valid specifier: {spec}")
+
+            if len(specifiers) > 0:
+                # Store in the node
+                item_contexts[index].regions = specifiers
+            else:
+                # Delete the whole interval or computation node
+                del node.items[index]
+
+            # Delete region AST node
+            del node.items[arg_index]
+
+            return node
+        else:
+            return self.generic_visit(node)
```

Comment:
Maybe split into a few more helpers? The main method here is a bit long.

Reply:
Definitely. This was addressed in 015de31.